src/clib/pio_darray.c

```diff
@@ -507,6 +517,7 @@ int PIOc_read_darray(const int ncid, const int vid, const int ioid,
     if (!(iodesc = pio_get_iodesc_from_id(ioid)))
         return PIO_EBADID;
 
+    /* ??? */
```

Comment:
For netcdf serial writes we collect the data on io nodes and then move that data one node at a time to the io master node and write (or read).   The buffer size on io task 0 must be as large as the largest used to accommodate this serial io method.